### PR TITLE
Added features to rcl_logging_spdlog

### PR DESCRIPTION
**Pull request by ahcorde**
Added features to `rcl_logging_spdlog`

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rcl_logging_spdlog/README.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 Package supporting an implementation of logging functionality using `spdlog`.
 
+The `rcl_logging` interface allows [rcl_logging_spdlog/logging_interface.h](include/rcl_logging_spdlog/logging_interface.h):
+ - initialize
```
Comment (hidmic) [inline rcl_logging_spdlog/README.md:6]: @ahcorde should we speak about this specific `spdlog` implementation instead? I'd think `rcl_logging` documentation already describes the interface in full.

File: rcl_logging_spdlog/README.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+ - log a message
+ - set the logger level
+ - shutdown
+
+Further still there are some useful abstractions and utilities:
+- Macros for controlling symbol visibility on the library
+ - [rcl_logging_spdlog/visibility_control.h](include/rcl_logging_spdlog/visibility_control.h)
```
Comment (hidmic) [inline rcl_logging_spdlog/README.md:13]: @ahcorde AFAIK these are for internal use only. Pretty much every package has a visibility control header.

Comment (ahcorde) [reply rcl_logging_spdlog/README.md:13]: other packages such as rclcpp consider this a [feature](http://docs.ros2.org/eloquent/api/rclcpp/index.html)

Comment (hidmic) [reply rcl_logging_spdlog/README.md:13]: It documents them as internal utilities (!= feature IMHO).

File: rcl_logging_spdlog/README.md [hunk 3]
```diff
@@ -2,15 +2,15 @@
 
 Package supporting an implementation of logging functionality using `spdlog`.
 
-The `rcl_logging` interface allows [rcl_logging_spdlog/logging_interface.h](include/rcl_logging_spdlog/logging_interface.h):
+[rcl_logging_spdlog](include/rcl_logging_spdlog/logging_interface.h):
```
Comment (hidmic) [inline rcl_logging_spdlog/README.md:5]: @ahcorde if we are going to keep a (probably duplicate) logging API enumeration, maybe something like: 
```suggestion
[rcl_logging_spdlog](include/rcl_logging_spdlog/logging_interface.h) logging interface implementation can:
```
would read better. Feel free to rephrase, I just made this one up off the top of my head.

**Pull request merged**